### DISCUSSION TITLE
Mapsforge map rotation

### DISCUSSION THREAD

**Discussion by Klaus-Tockloth** [devemux86/cruiser#99] [Q&A] [answered]
For me the most important (and still missing) feature is the ability to automatically rotate the map in the direction of movement. This helps with the orientation enormously. Cruiser-VTM has this feature, Cruiser-Mapsforge unfortunately does not. Locus Map also has this feature for Mapsforge maps.

**Comment by devemux86** [comment, marked as answer]
@Klaus-Tockloth thanks for the request!

Cruiser uses the original [Mapsforge](https://github.com/mapsforge/mapsforge) and [VTM (OpenGL)](https://github.com/mapsforge/vtm) map libraries.
(I am the developer of both map libraries)

Locus uses a modified Mapsforge or its own renderer?
@menion can better answer this.

---

Such map features must first be implemented in the map library.

If there is Mapsforge code available for map rotation or float zoom,
I will be happy to review and merge it into Mapsforge!

**Comment by 0709wiwiwi** [comment]
> automatically rotate the map in the direction of movement

Is this about the desktop version or is it about the Android app ? 
In the Cruiser Android app,if set the map turns correct in the direction of movement right ?
This can even inclusive map tilt and street names NOT staying upside down when turning.

**Reply by devemux86** [reply]
Android Cruiser when set Mapsforge as rendering engine in advanced map settings.

**Reply by 0709wiwiwi** [reply]
Where is that good for ?

**Reply by devemux86** [reply]
Many map themes are written in the [Mapsforge syntax](https://github.com/mapsforge/mapsforge/blob/master/resources/renderTheme.xsd) (OpenAndroMaps, Freizeitkarte).
VTM is compatible with the Mapsforge themes, but the result may not always be perfect.

Map themes like "Biker" or the other GL integrated themes are written in the [VTM syntax](https://github.com/mapsforge/vtm/blob/master/resources/rendertheme.xsd).
This means a native result that works better with OpenGL rendering.

**Comment by menion** [comment]
Just for confirmation: Locus Map uses a completely own system for work with maps and the MapsForge library is used "only" to draw bitmap map tiles from the vector map files. So the rotation is implemented in the app itself and is not inherited from the MF library. So I just confirm what @devemux86 wrote.
